packages/react-reconciler/src/ReactFiberHostContext.js

```diff
@@ -79,7 +82,12 @@ function pushHostContext(fiber: Fiber): void {
     rootInstanceStackCursor.current,
   );
   const context: HostContext = requiredContext(contextStackCursor.current);
-  const nextContext = getChildHostContext(context, fiber.type, rootInstance);
+  let type = fiber.type;
+  // For React event components, we get the typeof field (the symbol)
+  if (enableEventAPI && fiber.tag === EventComponent) {
+    type = type.$$typeof;
+  }
+  const nextContext = getChildHostContext(context, type, rootInstance);
 
   // Don't push this Fiber's context unless it's unique.
   if (context === nextContext) {
```

Comment:
If you create a separate pushHostContextForEvent function then for renderers where we call `getChildHostContextForEvent` and that's an identity function that just returns the same context, then this should be able to be DCE since it will know that the context is always the same.